app/controllers/services_controller.rb

```diff
@@ -25,6 +25,17 @@ def create
     end
   end
 
+  def destroy
+    service = Service.find_by_id(params[:id])
+    return flash[:notice] = t('common.not_found', name: 'Service') unless service.present?
+
+    event = DeleteServiceEvent.create(service: service, data: { name: service.name, entity_id: service.entity_id })
+    return flash[:error] = event.errors.full_messages.join(', ') if event&.errors.present?
```

Comment:
we don't need to this complicated, get inspired by the component deletion https://github.com/alphagov/verify-self-service/blob/ab9f4cd4f188f4ec31a4225d61373162364d1571/app/controllers/msa_components_controller.rb#L52

there's no reason for the event to be invalid as long as the service exists, so we should only need to check for the service existence and then fire the event.. all it does is deletion, no validation or anything...

Reply:
This was what I didn't on the first round must have misinterpreted you comment. And gone down this rabbit hole.  "and only if the event was valid" one of you first comments on friday

Reply:
it was a general comment about validity... sorry, my bad, i didn't mean the literal object's validity...